src/subscription/subscribe.js

```diff
@@ -32,10 +35,17 @@ import type { GraphQLFieldResolver } from '../type/definition';
 /**
  * Implements the "Subscribe" algorithm described in the GraphQL specification.
  *
- * Returns an AsyncIterator
+ * Returns a Promise<AsyncIterator | Error | ExecutionResult>
```

Comment:
Or maybe it's easier to read if this spells it out. Something like:

Returns a Promise which resolves to either an ExecutionResult (indicating failure to subscribe) or an AsyncIterator of ExecutionResult values, triggered by the underlying subscribed event stream.